pydpkg/__init__.py

```diff
@@ -38,80 +43,73 @@ class Dpkg(object):
     """Class allowing import and manipulation of a debian package file."""
 
     def __init__(self, filename=None):
-        self.headers = {}
-        if not isinstance(filename, basestring):
+
+        if not isinstance(filename, six.string_types):
             raise DpkgError('filename argument must be a string')
+
         if not os.path.isfile(filename):
-            raise DpkgError('filename "%s" does not exist', filename)
-        self.control_str, self._control_headers = self._process_dpkg_file(
-            filename)
-        for k in self._control_headers.keys():
-            self.headers[k] = self._control_headers[k]
+            raise DpkgError('filename "%s" does not exist' % filename)
+
+        self.control_str, self.headers = self._process_dpkg_file(filename)
```

Comment:
This changes self.headers from a dict to an `email.message.Message` object in py3 or a `rfc822.Message` object in py2, and to my intense annoyance the two different Message objects are not equivalent: rfc822.Message does not have a `__getitem__`:

```
>>> x.headers
<email.message.Message object at 0x11050dc50>
>>> x.headers['version']
'1:0.0.0-test'
```

...vs...

```
>>> x = Dpkg('/Users/n/testdeb_1:0.0.0-test_all.deb')
>>> x.headers
<rfc822.Message instance at 0x100da51b8>
>>> x['version']
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'Dpkg' object has no attribute '__getitem__'
```

Since `headers` is a public attribute, I'd rather not have its behavior change based on python version; I'd strongly prefer to keep interpolating it into a dict, ugly though it is. :(